src/util/with-props.js

```diff
@@ -8,7 +8,9 @@ const _syncingPropertyToAttribute = sym('_syncingPropertyToAttribute');
 export const _updateDebounced = sym('_updateDebounced');
 
 export function defineProps (Ctor) {
-  if (!Ctor[_definedProps]) {
+  if (Ctor[_definedProps]) {
+    return;
+  } else {
```

Comment:
just a nitpick.

Else is unnecessary and makes code less readable

consider:
```js
export function defineProps (Ctor) {
  if (Ctor[_definedProps]) {
    return
  }

  Ctor[_definedProps] = true;

  const { prototype } = Ctor;
  const props = normPropDefs(Ctor);

// ...

}